src/ui/json/JsonArray.js

```diff
@@ -8,40 +8,72 @@ import JsonBoolean from './JsonBoolean';
 import JsonNull from './JsonNull';
 
 class JsonArray extends Component<{
-  name: ?string,
+  depth: number,
   json: Array<mixed>,
 }> {
   static defaultProps = {
-    name: null,
+    depth: 0,
     json: [],
   };
 
   render() {
+    const currentIndentation = (
+      <React.Fragment>
+        {Array.from(new Array(this.props.depth).keys()).map((_) => {
+          return <React.Fragment>&nbsp; </React.Fragment>;
+        })}
+      </React.Fragment>
+    );
+    const nextIndentation = (
+      <React.Fragment>&nbsp; {currentIndentation}</React.Fragment>
+    );
     return (
-      <div className="jsonArray">
-        {this.props.name ? `"${this.props.name}": ` : ''}
+      <span className="jsonArray">
         {'['}
-        {this.props.json.map((value, index) => {
-          switch (typeof value) {
-            case 'object':
-              if (value === null) {
-                return <JsonNull key={index} />;
-              } else if (Array.isArray(value)) {
-                return <JsonArray json={value} key={index} />;
+        {this.props.json.length === 0 ? null : (
+          <ul>
+            {this.props.json.map((value, index) => {
+              let jsonContent;
+              switch (typeof value) {
+                case 'object':
+                  if (value === null) {
+                    jsonContent = <JsonNull />;
+                    break;
+                  } else if (Array.isArray(value)) {
+                    jsonContent = (
+                      <JsonArray json={value} depth={this.props.depth + 1} />
+                    );
+                    break;
+                  }
+                  jsonContent = (
+                    <JsonObject json={value} depth={this.props.depth + 1} />
+                  );
+                  break;
+                case 'number':
+                  jsonContent = <JsonNumber json={value} />;
+                  break;
+                case 'string':
+                  jsonContent = <JsonString json={value} />;
+                  break;
+                case 'boolean':
+                  jsonContent = <JsonBoolean json={value} />;
+                  break;
+                default:
+                  throw new Error('Invalid json property');
               }
-              return <JsonObject json={value} key={index} />;
-            case 'number':
-              return <JsonNumber json={value} key={index} />;
-            case 'string':
-              return <JsonString json={value} key={index} />;
-            case 'boolean':
-              return <JsonBoolean json={value} key={index} />;
-            default:
-              throw new Error('Invalid json property');
-          }
-        })}
+              return (
+                <li key={index}>
```

Comment:
@shihangw I set `key` attribute here, so React knows which child element is which after update.